staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -143,9 +146,30 @@ func (s *ServerRunOptions) Validate() []error {
 		errors = append(errors, fmt.Errorf("--max-resource-write-bytes can not be negative value"))
 	}
 
+	if err := validateHSTSHeader(s.HSTSHeaders); err != nil {
+		errors = append(errors, err)
+	}
 	return errors
 }
 
+func validateHSTSHeader(hstsHeaders []string) error {
+	// HSTS Headers format: Strict-Transport-Security:max-age=expireTime [;includeSubDomains] [;preload]
+	if len(hstsHeaders) != 0 {
+		for _, hstsHeader := range hstsHeaders {
+			if len(strings.TrimSpace(hstsHeader)) == 0 {
+				return fmt.Errorf("empty value in strict-transport-security-headers")
+			}
+			if hstsHeader != "includeSubDomains" && hstsHeader != "preload" {
+				maxAgeValue := strings.Split(hstsHeader, "=")
+				if len(maxAgeValue) != 2 || maxAgeValue[0] != "max-age" {
+					return fmt.Errorf("--strict-transport-security-headers invalid, must be of format: max-age=expireTime,includeSubDomains,preload")
```

Comment:
please link to the spec saying these are the only valid values.  The format appears more flexible to me https://tools.ietf.org/html/rfc6797#section-6.1 with some kind of a registry somewhere for extension.